### ISSUE TITLE
Remote validation deletes previous validation errors

### ISSUE THREAD

**Issue by florianengel** [jzaefferer/jquery-validation#1375]
If one element on a page is validated using the remote validation method, the validation errors of previous elements get lost.
Since I wasn't able to provide a working example on jsfiddle or jsbin, which actually does a remote call, I put the demo here: http://bit.ly/14Xt9Da

If you leave all 3 input elements empty, the required rule for the first and third element works correctly, since the remote validation doesn't get triggered. If you fill something in the second field, only the validation message for the third one shows up. (the true.php returns true for every input).
This behavior occurs also in the latest version 1.13.1. 

During the remote success handling the function prepareElement() gets called, which will then call this.reset();
```javascript
		reset: function() {
			this.successList = [];
			this.errorList = [];
			this.errorMap = {};
			this.toShow = $( [] );
			this.toHide = $( [] );
			this.currentElements = $( [] );
		},
```
Particularly the this.errorMap = {} seems to cause the problem here. Till the validator reaches the statement, it holds all the previous errors.

**Comment by jzaefferer** [comment]
Thank you for the report. The remote method isn't the only one causing issues like this. Calling `valid()` on any element directly also does that. A proper solution would track each element individually, instead of dumping everything into a single object and resetting that together.

**Comment by c-krani** [comment]
Recently we encountered the same issue. I don't unterstand why prepareElement() gets called in the success-Method of the remote validation. None of the other valdiation methods does. 
Anybody knows the reason?

**Comment by jzaefferer** [comment]
I'm sorry for the lack of activity on this issue. Instead of leaving it open any longer, I decided to close old issues without trying to address them, to longer give the false impression that it will get addressed eventually, especially after several years with no activity. It doesn't mean I'm abandoning the project, just that I'm unable to work through 200+ open issues with the little time I can afford to spend on this project.

To the reporter (or anyone else interested in this issue): If you're still affected by the same issue, please consider opening a new issue, with a testpage that demonstrates the issue with a current version of the plugin. Even better, make an attempt to fix the issue yourself, and improve the project by sending a pull request. This may seem daunting at first, but you'll likely learn some useful skills that you can apply elsewhere as well. And you can help keep this project alive. [We've documented how to do these things, too](https://github.com/jzaefferer/jquery-validation/blob/master/CONTRIBUTING.md). A patch is worth a thousand issues!
